CRM/MembershipExtras/Hook/PostProcess/MembershipTypeSetting.php

```diff
@@ -48,13 +48,13 @@ private function updateMembershipTypeSettings() {
       $this->settings = [];
     }
 
-    if (!empty($this->form->_id)) {
-      $annualProRataCalculationValue = $this->form->_submitValues['membership_type_annual_pro_rata_calculation'];
-      $this->settings[$this->form->_id]  = [
-        'membership_type_annual_pro_rata_calculation' => $annualProRataCalculationValue,
-      ];
-      Civi::settings()->set(SettingsManager::MEMBERSHIP_TYPE_SETTINGS_KEY, $this->settings);
-    }
+    $membershipTypeId = !empty($this->form->_id) ? $this->form->_id : $this->getMembershipTypeId();
```

Comment:
Should we move the condition to `getMembershipTypeId` method.

Reply:
@ahed-compucorp  I was thinking about this too, but I do not want to call the API when we edit the membership form as the ID would already exist.

Reply:
I mean like this :


    private function getMembershipTypeId() {
        if (!empty($this->form->_id)) {
           $membershipTypeId = $this->form->_id; 
        } else { 
          $membershipTypeId = civicrm_api3('MembershipType', 'getsingle', [
            'return' => ['id'],
            'name' => $this->form->exportValues()['name'],
          ])['id'];
        }
        return $membershipTypeId;
    }